### PR TITLE
Membership schema

### PR DESCRIPTION
**Pull request by metmirr**
This PR:
- Adds mappings for membership module
- Fixes `forEach` async/await problem 
- Use `bn.js` for large numbers. e.g block number, event index

### CODE REVIEW

**Review by bedeho — APPROVED** [review]
LGTM!

File: query-node/substrate-query-framework/index-builder/src/db/helper.ts [hunk 1]
```diff
@@ -10,6 +10,9 @@ import { DeepPartial } from 'typeorm';
  * @param entity: DeepPartial<T>
  */
 export function fillRequiredWarthogFields<T>(entity: DeepPartial<T>): DeepPartial<T> {
+  // Modifying an existing entity so do not add warthog fields
+  if (entity.hasOwnProperty('id')) return entity;
```
Comment (dzhelezov) [inline query-node/substrate-query-framework/index-builder/src/db/helper.ts:14]: What about the extra fields like `createdAt` ? Do we still need them in this case?

File: query-node/substrate-query-framework/cli/src/templates/index-builder-entry.mst [hunk 2]
```diff
@@ -122,7 +122,9 @@ async function main() {
       "WS_PROVIDER_ENDPOINT_URI environment variable is not set! Make sure you set it in your .env file or system wide."
     );
   }
-
+  
+  // Required for warthog models.
+  dotenv.config({ path: '../graphql-server/.env' });
```
Comment (dzhelezov) [inline query-node/substrate-query-framework/cli/src/templates/index-builder-entry.mst:127]: This makes the indexer dependent on the graph server, isn't it? What env properties do we need here apart from the DB?

Comment (metmirr) [reply query-node/substrate-query-framework/cli/src/templates/index-builder-entry.mst:127]: Yes. Just `WARTHOG_APP_HOST` and `WARTHOG_APP_PORT` actually, DB env is the same with warthog. Should I add them to indexer's `.env` or?

File: query-node/substrate-query-framework/index-builder/src/db/helper.ts (hunk 1 above)
Comment (metmirr) [reply query-node/substrate-query-framework/index-builder/src/db/helper.ts:14]: This guard is for updating existing db records. We don't need those extra fields but warthog `BaseModel` adds them :(

File: query-node/substrate-query-framework/cli/src/templates/index-builder-entry.mst (hunk 2 above)
Comment (dzhelezov) [reply query-node/substrate-query-framework/cli/src/templates/index-builder-entry.mst:127]: Yeah, I think we already have `templates/dotenv.mst` for that don't we?

File: query-node/substrate-query-framework/index-builder/src/db/helper.ts (hunk 1 above)
Comment (dzhelezov) [reply query-node/substrate-query-framework/index-builder/src/db/helper.ts:14]: Ah ok. I thought that this check is like 'if the user has provided an ID field in the schema, use it'

File: query-node/substrate-query-framework/cli/src/templates/index-builder-entry.mst (hunk 2 above)
Comment (dzhelezov) [reply query-node/substrate-query-framework/cli/src/templates/index-builder-entry.mst:127]: Probably a more serious problem here is that we somehow need to import the model files from graphql-server. I remember we already had this issue, how do we solve it?

Comment (metmirr) [reply query-node/substrate-query-framework/cli/src/templates/index-builder-entry.mst:127]: Yeah, we have discussed importing models but we don't have a solution yet.

File: query-node/substrate-query-framework/index-builder/src/db/helper.ts (hunk 1 above)
Comment (metmirr) [reply query-node/substrate-query-framework/index-builder/src/db/helper.ts:14]: 👍

File: query-node/substrate-query-framework/cli/src/templates/index-builder-entry.mst (hunk 2 above)
Comment (dzhelezov) [reply query-node/substrate-query-framework/cli/src/templates/index-builder-entry.mst:127]: Ok, it's out of scope for this PR, but here's an issue for the future reference: https://github.com/Joystream/joystream/issues/723

Comment (metmirr) [reply query-node/substrate-query-framework/cli/src/templates/index-builder-entry.mst:127]: 👍

**Pull request merged**